Created rough profile header for use on guide and user profile pages.

RyuGuardian opened the pull request.

RyuGuardian committed:
Created rough profile header for use on guide and user profile pages.

RyuGuardian committed:
Removed unneeded line.

Octember commented on app/components/ProfileHeader/index.js:
```diff
@@ -0,0 +1,71 @@
+      <ProfileImg src={props.url} className="mt-4"
+        alt={props.guide ? 'guide logo' :'profile image'}
+      />
+      <h1 className="h2 text-info">{props.name}</h1>
```
h1 className h2? I don't think that h2 classname does anything

RyuGuardian replied:
It styles it like an h2 (Bootstrap's h2). It's a simple way to style text to a certain size and give it margins without messing with the HTML semantics or making a new styled component.

Octember replied:
Sorry I'm still confused, what do you mean "messing with HTML semantics"? Do you mean for like SEO, crawlers?

RyuGuardian replied:
I don't have much knowledge on SEO and crawlers, but yeah, I think this is along the lines of the principles "HTML elements matter" (when considering semantic HTML) and "class names don't matter" (which seems to be the Bootstrap approach to styling?).

I'm trying to use the advantages Bootstrap gives me and give thought into the types of elements used in a page (I think it's also important for screen readers).

Octember replied:
Interesting. Good thinking, I think you're probably right! Big props

RyuGuardian replied:
Thanks!

Octember commented on app/components/ProfileHeader/index.js:
```diff
@@ -0,0 +1,71 @@
+      {
+        // Only creates a location/summary element if one exists
+        (props.location || props.summary) &&
+        <h2 className="h5 mb-3 text-muted">{props.location ? props.location : props.summary}</h2>
```
h5 classname? What's the point of that?

RyuGuardian replied:
Font sizing (see previous comment). My logic here is that the heading is significant enough to be an h2 (rather important), but since the mock-up showed a smaller font, it was simpler to use Bootstrap's h5 class (as recommended) rather than make a new styled component.

Octember commented on app/components/ProfileHeader/index.js:
```diff
@@ -0,0 +1,71 @@
+          Object.keys(props.stats).map((stat, i, arr) => {
+            return (
+              <li className="list-inline-item" key={i}>
+                <p className="h3 text-info">{props.stats[stat]}</p>
```
`<h3>` instead of classname h3, right?

RyuGuardian replied:
I've looked up many times how h1-h6 should be used, and there doesn't seem to be a definitive guide, but I do think a heading should be a title rather than data like a number. Since the numbers here are large, they need the styling of a heading, though, and that's all the class is for.

Octember commented on app/components/ProfileHeader/index.js:
```diff
@@ -0,0 +1,71 @@
+    <BorderBottomDiv>
+      <ul className="list-inline row m-2 justify-content-around">
+        {
+          Object.keys(props.stats).map((stat, i, arr) => {
```
What is `arr`? Looks unused, can we remove?

RyuGuardian replied:
Oops, one of those things I thought I was gonna use then didn't.

Octember commented on app/components/ProfileHeader/index.js:
```diff
@@ -0,0 +1,71 @@
+                <p className="small text-muted mb-0">
+                  {/* Capitalize stat name and add space before camelcase or snakecase character (or number) */}
+                  { stat.slice(0, 1).toUpperCase() + stat.slice(1)
+                    .replace(/(_[a-z])|[_]?([A-Z0-9])/g, (match, p1, p2) => {
```
This is way too complicated to render inline. If we needed this functionality I'd want it to be broken out into a separate function

Octember replied:
Instead of this, let's make the `stats` object be of a shape, e.g.

```
stats: PropTypes.objectOf(PropTypes.shape({
    count: PropTypes.number,
    label: PropTypes.string,
  })
```

In general, I don't want any complicated logic on the frontend. It will sometimes be needed, but encoding business logic here is a recipe for bugs down the line IMO

RyuGuardian replied:
OK. Wasn't sure what would be best to do here, so just went with my first thought and finished it to get it done and turn in (I also like working with RegEx).

Octember commented on app/containers/GuideProfilePage/index.js:
```diff
@@ -36,7 +49,8 @@ export class GuideProfilePage extends React.Component { // eslint-disable-line r
                 {/* Mobile/tablet view */}
-                mobile header
+                {/* Currently using dummy profile */}
+                <ProfileHeader guide {...sampleProfile} />
```
Can you put it on desktop too?

RyuGuardian replied:
NP; no styling for desktop is set up, though.

Octember commented on app/containers/GuideProfilePage/index.js:
```diff
@@ -11,15 +11,28 @@ import { Helmet } from 'react-helmet';
+import Header from 'components/Header';
+import ProfileHeader from 'components/ProfileHeader';
+
+const sampleProfile = {
```
The API already exists for this - you don't need to do this now but we'll need to integrate that call into the props

RyuGuardian replied:
Yeah, I did this right before you announced the API. I'll integrate it with the Guide Page before moving on to something else.

Octember replied:
I made a separate ticket for it, feel free to do that as a separate PR

Octember reviewed (changes requested):
The page looks great. Some philosophical requests. I'll make a ticket to plumb the data into this.

RyuGuardian committed:
Requested changes from pull request
 - Removed logic from profile page and added shape props for labels and counters
 - Added component to desktop display

RyuGuardian committed:
Merge master into profile-header

RyuGuardian commented:
Things should be good to go now after review and merging. Will work on integrating Guide API now.

Octember commented on app/containers/GuideProfilePage/index.js:
```diff
@@ -11,15 +11,28 @@ import { Helmet } from 'react-helmet';
+  url: 'http://travelchair.com/images/blockexpeditions.png',
+  location: 'Winthrop, WA',
+  stats: [
+    { label: 'Followers', count: 4 },
```
nice

Octember commented:
Looks good! Gonna merge this so you have a clean state for the next piece

nice work dude
